### PR TITLE
[Feature] OD-8 Integrated data source provider to the Chat component

### PR DESCRIPTION
**Pull request by YaroslavOvdii**
@sofiiakvasnevska 

## Issue
https://weboo.atlassian.net/browse/OD-8

## Description
Integrated data source provider to the Chat component

## Screenshots/screencasts
https://share.getcloudapp.com/DOu9v9Ox

## Backward compatibility
This change is fully backward compatible.

## Reviewers
@upplabs-alex-levchenko

### CODE REVIEW

File: interface.html [hunk 1]
```diff
@@ -14,17 +14,7 @@
...
-        </div>
-        <div class="col-sm-8">
-          <select name="dataSource" id="contactsList" data-label="select" class="hidden-select form-control" disabled>
-            <option value="none">-- Select a data source</option>
-            <option disabled>------</option>
-            <option value="new">Create a new data source</option>
-            <option disabled>------</option>
-          </select>
-          <small id="manage-data" class="hidden"><a href="#">Edit data source</a></small>
+        <div id="dataSourceProvider" class="col-sm-8 ui-sortable" >
```
Comment (squallstar) [inline interface.html:17]: there's an empty space to remove

File: js/interface.js [hunk 2]
```diff
@@ -79,38 +77,50 @@ $('#show-full-name-field').on('click', function() {
...
+      name: 'Chat data for' + Fliplet.Env.get('appName'),
+      entries: [],
+      columns: []
+    },
+    accessRules: [
+      {
+        allow: 'all',
+        enabled: true,
+        type: [
+          'select'
```
Comment (squallstar) [inline js/interface.js:98]: I believe chat also needs to be able to update the data source records.

**Review by squallstar — CHANGES_REQUESTED** [review]

File: js/interface.js (hunk 2 above)
Comment (YaroslavOvdii) [reply js/interface.js:98]: @squallstar Should chat only update and ready record or adding them as well?

Comment (squallstar) [reply js/interface.js:98]: just read and update

Comment (YaroslavOvdii) [reply js/interface.js]: @squallstar Added

File: interface.html (hunk 1 above)
Comment (YaroslavOvdii) [reply interface.html]: @squallstar fixed

**Review by squallstar — APPROVED** [review]

**Pull request merged**